### PR TITLE
add support for annotations

### PR DESCRIPTION
**Pull request by jtwires**
* add methods for creating/retrieving/removing annotations
* parse bracket- and semicolon-style annotations when loading PGNs
* output bracket-style annotations when producing PGNs

### CODE REVIEW

**Comment by jhlywa** [comment]
Thanks Jake.  I'll review this over the next few days.

**Comment by jhlywa** [comment]
Thanks @jtwires.  It's clear you put a lot of work into this PR.  I've got a few questions and suggestions before I merge.

1.  I can see the use-case arising where a user would want to remove all annotations from a PGN.  How would you feel about adding an explicit `clear_annotation()` function that could be called with an optional argument to clear all annotations (e.g. `clear_annotation({target: 'game'})` or `clear_annotation({all: true})` or something similar, picking good names is hard).  Returning the annotation that was cleared (or null) would be a nice touch too.

2. The [PGN spec](https://www.chessclub.com/help/PGN-spec) refers to these annotations as comments; basically anything in `{ ... }` or `;...`, while using the term annotation for Recursive Annotation Variations.  Could you change the API to use the term `comments` instead?

3.  I've added a few inlines comments in the PR itself.

Many people have been requesting this capability.  Thanks again for your contribution.

File: README.md [hunk 1]
```diff
@@ -58,6 +58,61 @@ const chess = new Chess(
...
+chess.pgn()
+// -> "1. e4"
+```
+
+### .annotation()
+
+Retrieve the annotation for the current position, if it exists.
+
+```js
+const chess new Chess()
```
Comment (jhlywa) [inline README.md:86]: Minor type `const chess = new Chess()`

File: chess.js [hunk 2]
```diff
@@ -1541,14 +1615,46 @@ var Chess = function(fen) {
         }
       }
 
+      var to_hex = function(string) {
```
Comment (jhlywa) [inline chess.js:1618]: Can you add a comment of too here to explain what you're doing with hex and URI encoding?  It's not immediately clear to me.  Thanks

**Comment by jtwires** [comment]
cool, `comment` it is..

**Comment by jhlywa** [comment]
0.11.0 has been pushed to npm.  I'll gen up a changelog sometime over the weekend.

**Pull request merged**